packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -103,15 +103,8 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
         this.snapshotUrl = this.odspResolvedUrl.endpoints.snapshotStorageUrl;
         this.attachmentPOSTUrl = this.odspResolvedUrl.endpoints.attachmentPOSTStorageUrl;
         this.attachmentGETUrl = this.odspResolvedUrl.endpoints.attachmentGETStorageUrl;
-
-        this.odspSummaryUploadManager = new OdspSummaryUploadManager(
-            this.snapshotUrl,
-            getStorageToken,
-            logger,
-            epochTracker,
-            !!this.hostPolicy.sessionOptions?.forceAccessTokenViaAuthorizationHeader,
-            this.relayServiceTenantAndSessionId,
-        );
+        this.odspSummaryUploadManagerPromise = import("./getOdspSummaryUploadManager").then(
```

Comment:
@christiango - can you please glance? I'm not sure how this impacts final packaging, what info we should be providing to consumers of FF repo such that they can make a call on how package such things (for example, maybe multiple chunks like this and socket.io should be grouped together into single bundle).
Maybe it's all fine and that's my lack of knowledge of tooling :)

Reply:
Then answer is "it depends" unfortunately. We should see how many new files the different approaches here webpack would spit out after accounting for the fact that webpack will split out "common" js modules between many async imports into "common chunks" to avoid duplication.

So if the socket.io chunk doesn't have a lot of overlap in terms of dependencies with what we are doing here, it probably won't matter. 

What I recommend doing is having a look at one of the demo webpack 5 apps  using odsp driver in the FF repo spits out with both approaches in terms of the number of files. There is some non-trivial overhead for the browser for each individual JS file, so the question we want to understand is "does this result in just 1 new JS file in webpack, or 10)

Reply:
Can we control it? I.e. without putting code into the same file, can we instruct bundler to group two chunks into same bundle? This might be viable strategy if there is obvious partitioning based on priority, i.e. "all code needed by summarizer - P4", "all code needed to connect to socket - P2", etc.

Reply:
It's challenging, especially with the fact that different bundlers behave very differently (esbuild vs webpack for example). One thing you can do to group things together is to create modules like p2.js and p4.js that just re-export these pieces, but that isn't guaranteed to do what we want.